### ISSUE TITLE
Github API: mergin PR #122 ... panic: "" is not a valid Git SHA

### ISSUE THREAD

**Issue by lud-wj** [git-town/git-town#3113]
Hi, 

I had this bug again. 

```
[feat/mybranch] git fetch --prune --tags

[feat/mybranch] git checkout main
Basculement sur la branche 'main'
Votre branche est à jour avec 'origin/main'.

[main] git merge --squash feat/mybranch
Mise à jour 5df78ef..8763707
Fast-forward
Validation compressée -- HEAD non mise à jour
 some file |  41 ++++++++++++-
 some file |   2 +
 some file |   2 +-
 some file |  24 ++++++++
 some file |  13 -----
 some file |   2 +
 some file | 278 ++++++++++++++++++++++++++++++++++++++++++++++------------------------------------------
 some file |  16 +-----
 some file |  23 +++++---
 some file |  43 +++++++++++---
 some file |   9 ++-
 some file |   2 +-
 some file | 205 ++++++++++++++++++++++++++++++++++++-----------------------------
 some file |  23 ++++++++
 some file | 112 +++++++++++++++++-------------------
 some file |  10 ++++
 16 files changed, 473 insertions(+), 332 deletions(-)

[main] git commit
[main 0f8d44d] feat: My super ship message
 16 files changed, 473 insertions(+), 332 deletions(-)

[main] git reset --hard HEAD~1
HEAD est maintenant à 5df78ef Merge pull request #123 from stuff

GitHub API: merging PR #122 ... panic: "" is not a valid Git SHA

goroutine 1 [running]:
github.com/git-town/git-town/v11/src/domain.NewSHA({0x0?, 0x0?})
        D:/a/git-town/git-town/src/domain/sha.go:19 +0x88
github.com/git-town/git-town/v11/src/hosting/github.(*Connector).SquashMergeProposal(0xc00037dd50, 0x7a, {0xc000844680, 0x3b})
        D:/a/git-town/git-town/src/hosting/github/connector.go:76 +0x1aa
github.com/git-town/git-town/v11/src/vm/opcode.(*ConnectorMergeProposal).Run(0xc000776910, {0xc0003e4000, {0xccd2b0, 0xc00037dd50}, 0xc0001cd710, 0xc0003e4010, 0xc0001d1720, 0xc000319e20})
        D:/a/git-town/git-town/src/vm/opcode/connector_merge_proposal.go:70 +0x1b7
github.com/git-town/git-town/v11/src/vm/interpreter.Execute({0xc0007d4370, 0xc0001d1720, {0xccd2b0, 0xc00037dd50}, 0x0, {0xc000026ac0, 0x1b}, {{0xc00033a500, 0xa, 0x10}, ...}, ...})
        D:/a/git-town/git-town/src/vm/interpreter/execute.go:26 +0x27b
github.com/git-town/git-town/v11/src/cmd.executeShip({0x113ee00, 0x0, 0x0}, {0x0, 0x0}, 0x0)
        D:/a/git-town/git-town/src/cmd/ship.go:101 +0x458
github.com/git-town/git-town/v11/src/cmd.shipCmd.func1(0xc0001ded00?, {0x113ee00, 0x0, 0x0})
        D:/a/git-town/git-town/src/cmd/ship.go:62 +0x72
github.com/spf13/cobra.(*Command).execute(0xc0001d5b00, {0x113ee00, 0x0, 0x0})
        D:/a/git-town/git-town/vendor/github.com/spf13/cobra/command.go:983 +0xabc
github.com/spf13/cobra.(*Command).ExecuteC(0xc000004300)
        D:/a/git-town/git-town/vendor/github.com/spf13/cobra/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
        D:/a/git-town/git-town/vendor/github.com/spf13/cobra/command.go:1039
github.com/git-town/git-town/v11/src/cmd.Execute()
        D:/a/git-town/git-town/src/cmd/core.go:33 +0x505
main.main()
        D:/a/git-town/git-town/main.go:22 +0x25
```

I remember a bug where `#` would cause problems in the logs.

Is there anything I can do for Git to ignore github stuff from the remote? I would be fine with a workaround.

**Comment by kevgo** [comment]
Sorry for the trouble Ludovic. I added this check to detect edge cases where Git Town receives non-expected server responses, and it looks like you hit such a case. Are you using a hosted GitHub Enterprise server by any chance?

In any case, the good news is that I have removed this check in https://github.com/git-town/git-town/pull/3114. The data it checks isn't used anymore thanks to other cleanup done a while ago. So this problem should be in the past now.

I have released [v12.0.1](https://github.com/git-town/git-town/releases/tag/v12.0.1) with a fix for this bug. Please give it a spin and reach out if you run into this or any other problems. Thanks! 🙏

**Comment by lud-wj** [comment]
Hello, yes I am using the Github organization for my company but I am not sure if it is hosted (I didn't even know that existed).

Thank you I'll try the new release ! 

Keep up the good work, git town rocks :°D
